azure-kusto-ingest/test/e2eTest.js

```diff
@@ -0,0 +1,170 @@
+    new testDataItem("json.gz with mapping", getTestResourcePath("dataset.json.gz"), 2, ingestionPropertiesWithColumnMapping, false)];
+
+var currentCount = 0;
+
+describe('E2E Tests', function () {
+    describe('SetUp', function () {
+        it('Drop table if exists', function (done) {
+            queryClient.execute(databaseName, `.drop table ${tableName} ifexists`, (err, results) => {
+                if (err) assert.fail("Failed to create table");
+                done();
+            });
+        });
+
+        it('Create table', function (done) {
+            queryClient.execute(databaseName, `.create table ${tableName} ${tableColumns}`, (err, results) => {
+                if (err) assert.fail("Failed to create table");
+                done();
+            });
+        });
+
+        it('Create table ingestion mapping', function (done) {
+
+
+            queryClient.execute(databaseName, `.create-or-alter table ${tableName} ingestion json mapping '${mappingName}' '${mapping}'`, (err, results) => {
+                if (err) assert.fail("Failed to create table ingestion mapping");
+                done();
+            });
+        });
+    });
+
+    describe('ingestClient', function () {
+        it('ingestFromFile', async function () {
+            for (let item of testItems) {
+                await new Promise((resolve) => {
+                    ingestClient.ingestFromFile(item.path, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+
+        it('ingestFromStream', async function () {
+            for (let item of testItems) {
+                let stream = fs.createReadStream(item.path);
+                if (item.path.endsWith('gz')) {
+                    stream = new StreamDescriptor(stream, null, CompressionType.GZIP);
+                }
+                await new Promise((resolve) => {
+                    ingestClient.ingestFromStream(stream, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
+                        resolve();
+                    });
+                });
+                await assertRowsCount(item);
+            }
+        });
+    });
+
+    describe('StreamingIngestClient', function () {
+        it('ingestFromFile', async function () {
+            for (let item of testItems.filter(item => item.testOnstreamingIngestion)) {
+                await new Promise((resolve) => {
+                    streamingIngestClient.ingestFromFile(item.path, item.ingestionProperties, async (err, results) => {
+                        if (err) assert.fail(`Failed to ingest ${item.description}`);
```

Comment:
For readability in case of failure, should we add to the assertion message the test's name? Also in assertRowCount

Reply:
we have it from the test framework

Reply:
I know, but is it convenient enough? I don't know this framework

Reply:
convenient+